linode_api4/linode_client.py

```diff
@@ -402,6 +402,133 @@ def __call__(self):
         p = Profile(self.client, result["username"], result)
         return p
 
+    def trusted_devices(self):
+        """
+        Returns the Trusted Devices on your profile.
+
+        API Documentation: https://www.linode.com/docs/api/profile/#trusted-devices-list
+
+        :returns: A list of Trusted Devices for this profile.
+        :rtype: PaginatedList of TrustedDevice
+        """
+        return self.client._get_and_filter(TrustedDevice)
+
+    def user_preferences(self):
+        """
+        View a list of user preferences tied to the OAuth client that generated the token making the request.
+        """
+
+        result = self.client.get(
+            "{}/preferences".format(Profile.api_endpoint), model=self
+        )
+
+        return MappedObject(**result)
+
+    def security_questions(self):
+        """
+        Returns a collection of security questions and their responses, if any, for your User Profile.
+        """
+
+        result = self.client.get(
+            "{}/security-questions".format(Profile.api_endpoint), model=self
+        )
+
+        return MappedObject(**result)
+
+    def security_questions_answer(self, questions):
+        """
+        Adds security question responses for your User. Requires exactly three unique questions.
+        Previous responses are overwritten if answered or reset to null if unanswered.
+
+        Example question:
+        {
+            "question_id": 11,
+            "response": "secret answer 3"
+        }
+        """
+
+        if len(questions) != 3:
+            raise ValueError("Exactly 3 security questions are required.")
+
+        params = {"security_questions": questions}
+
+        result = self.client.post(
+            "{}/security-questions".format(Profile.api_endpoint),
+            model=self,
+            data=params,
+        )
+
+        return MappedObject(**result)
+
+    def user_preferences_update(self, **preferences):
+        """
+        Updates a user’s preferences.
+        """
+
+        result = self.client.put(
+            "{}/preferences".format(Profile.api_endpoint),
+            model=self,
+            data=preferences,
+        )
+
+        return MappedObject(**result)
+
+    def phone_number_delete(self):
```

Comment:
May worth returning a boolean value to indicate if the api call succeeds or not. If not, we can raise an error. We can add a description to the return type in the method comment. Same for the a following couple of `post` methods.

Reply:
Fixed!